auth_backends/backends.py

```diff
@@ -244,20 +244,33 @@ class EdXOAuth2(EdXBackendMixin, BaseOAuth2):
 
     @property
     def logout_url(self):
-        return '{}?client_id={}&redirect_url={}'.format(
-            self.end_session_url(),
-            self.setting('KEY'),
-            self.setting('LOGOUT_REDIRECT_URL')
-        )
+        if self.setting('LOGOUT_REDIRECT_URL'):
+            return '{}?client_id={}&redirect_url={}'.format(
+                self.end_session_url(),
+                self.setting('KEY'),
+                self.setting('LOGOUT_REDIRECT_URL')
+             )
+        else:
+            return self.end_session_url()
 
     def authorization_url(self):
-        return '{}/oauth2/authorize'.format(self.setting('URL_ROOT'))
+        url_root = self.setting('PUBLIC_URL_ROOT')
+
+        if not url_root:
+            url_root = self.setting('URL_ROOT')
+
+        return '{}/oauth2/authorize'.format(url_root)
 
     def access_token_url(self):
         return '{}/oauth2/access_token'.format(self.setting('URL_ROOT'))
 
     def end_session_url(self):
-        return '{}/logout'.format(self.setting('URL_ROOT'))
+        url_root = self.setting('PUBLIC_URL_ROOT')
```

Comment:
Seems like this logic should be broken out into a method to DRY it out - since it's used twice?